File: src/main/java/org/blacksun/graph/algorithms/PathFindingAlgorithmFactory.java
package org.blacksun.graph.algorithms;

import org.blacksun.graph.channel.Channel;
import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface PathFindingAlgorithmFactory {
    PathFindingAlgorithm getAlgorithm(@NotNull List<GraphNode> nodes,
                                      @NotNull List<Channel> links);
}

File: src/main/java/org/blacksun/graph/algorithms/PathFindingAlgorithmFactory.kt
package org.blacksun.graph.algorithms

import org.blacksun.graph.channel.Channel
import org.blacksun.graph.node.GraphNode

interface PathFindingAlgorithmFactory {
    fun getAlgorithm(nodes: List<GraphNode>,
                     links: List<Channel>): PathFindingAlgorithm
}
